include/libtransistor/cpp/types.hpp

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include<libtransistor/types.h>
+
+#include<neither/neither.hpp>
+
+#include<stdexcept>
+#include<string>
+
+namespace Transistor {
+
+struct Unit {};
```

Comment:
Does something like this already exist in STL?

Reply:
http://en.cppreference.com/w/cpp/utility/variant/monostate

not sure if it's standard practice to actually use it independently of std::variant though